source/kraken/apply_disruption.cpp

```diff
@@ -190,11 +202,94 @@ struct add_impacts_visitor : public apply_impacts_visitor {
+                continue;
+            }
+
+            nt::ValidityPattern new_vp{vj->validity_patterns[rt_level]->beginning_date};
+
+            for(const auto& period : impact->application_periods) {
+                new_vp.days |= vj->get_vp_of_sp(*stop_point, rt_level, period).days;
+            }
+
+            if(new_vp.days.none()){
+                // The vj doesn't stop at the impacted stop_point during all the given impact periods
+                continue;
+            }
+            new_vp.days >>= vj->shift;
+            vj_vp_pairs.emplace_back(vj, new_vp);
+        }
+
+        for (auto& vj_vp : vj_vp_pairs) {
+            std::vector<nt::StopTime> new_stop_times;
+            const auto* vj = vj_vp.first;
+            auto& new_vp = vj_vp.second;
+
+            for (const auto& st : vj->stop_time_list) {
+                if (st.stop_point == stop_point) {
+                    continue;
+                }
+                nt::StopTime new_st = st.clone();
+                new_st.arrival_time = st.arrival_time + ndtu::SECONDS_PER_DAY * vj->shift;
+                new_st.departure_time = st.departure_time + ndtu::SECONDS_PER_DAY * vj->shift;
+                new_stop_times.push_back(std::move(new_st));
+            }
+
+            auto mvj = vj->meta_vj;
+            mvj->impacted_by.push_back(impact);
+
+            auto nb_rt_vj = mvj->get_vjs_at(rt_level).size();
+            std::string new_vj_uri = mvj->uri + ":" +
+                    type::get_string_from_rt_level(rt_level) + ":" +
+                    std::to_string(nb_rt_vj) + concatenate_impact_uris(*mvj);
+
+            new_vp.days = new_vp.days & (vj->validity_patterns[rt_level]->days >> vj->shift);
+
+            auto* new_vj = mvj->create_discrete_vj(new_vj_uri,
+                    rt_level,
+                    new_vp,
+                    vj->route,
+                    std::move(new_stop_times),
+                    pt_data);
+            
+            if (! mvj->get_base_vj().empty()) {
+                new_vj->physical_mode = mvj->get_base_vj().at(0)->physical_mode;
+                new_vj->name = mvj->get_base_vj().at(0)->name;
+            } else {
+                // If we set nothing for physical_mode, it'll crash when building raptor
+                new_vj->physical_mode = pt_data.physical_modes[0];
+                new_vj->name = new_vj_uri;
+            }
+            new_vj->physical_mode->vehicle_journey_list.push_back(new_vj);
+            new_vj->company = vj->company;
+            new_vj->vehicle_journey_type = vj->vehicle_journey_type;
+            new_vj->odt_message = vj->odt_message;
+            new_vj->_vehicle_properties = vj->_vehicle_properties;
+
+        }
     }
 
     void operator()(nt::StopArea* stop_area) {
-        LOG4CPLUS_INFO(log, "Disruption on stop area:" << stop_area->uri << " is not handled");
+        for (auto* stop_point: stop_area->stop_point_list) {
```

Comment:
minor: I think it would be nice to add back the `log_start_action(stop_area->uri);` (on stop_point too)

since it's a potential bug nest, adding debug log can help